railties/CHANGELOG.md

```diff
@@ -4,6 +4,32 @@
 
     *Jonathan Hefner*
 
+*   Add `Rails.application.url` and `config.application_url`
+
+    Before it you had to set static application-url in different places.
+
+    ```ruby
+    Rails.application.configure do
+      # Add top-level Rails.configuration.x.url for canonical reference
+      config.x.url = URI("http://myapplication.com")
+
+      # These options will be used as a starting point when generating URLs
+      config.action_controller.default_url_options = { host: 'myapplication.com' }
+      config.action_mailer.default_url_options     = { host: 'myapplication.com' }
+    end
+    ```
+
+    With this change, you only have to set it once.
+
+    ```ruby
+    Rails.application.configure do
+      # Add top-level Rails.application.url for canonical reference
+      config.application_url = "http://myapplication.com"
```

Comment:
Example should assume https, for most likely success if a user autopilots their copy & paste.